src/refdb_fs.c

```diff
@@ -1099,22 +1099,27 @@ static int packed_write(refdb_fs_backend *backend)
 static int packed_delete(refdb_fs_backend *backend, const char *ref_name)
 {
 	size_t pack_pos;
-	int error;
+	int error, found = 0;
 
 	if ((error = packed_reload(backend)) < 0)
 		goto cleanup;
 
-	/* If a packed reference exists, remove it from the packfile and repack */
 	if ((error = git_sortedcache_wlock(backend->refcache)) < 0)
 		goto cleanup;
 
-	if (!(error = git_sortedcache_lookup_index(
-			&pack_pos, backend->refcache, ref_name)))
+	/* If a packed reference exists, remove it from the packfile and repack if necessary */
+	error = git_sortedcache_lookup_index(&pack_pos, backend->refcache, ref_name);
+	if (error == 0) {
 		error = git_sortedcache_remove(backend->refcache, pack_pos);
+		found = 1;
+	}
+	if (error == GIT_ENOTFOUND)
+		error = 0;
 
 	git_sortedcache_wunlock(backend->refcache);
 
-	error = packed_write(backend);
```

Comment:
It's not only about clobbering error values, is it? Previously, we would've tried to write the packed file even in cases where we didn't remove any reference at all, right?

Reply:
Yes, from a functional POV it's slightly different in that the packed file will not be *rewritten*. AFAIR, this was also some kind of performance improvement, in that a packed-ref lookup failure when deleting would cause a "spurious" rewrite of an innocent file (but please double-check me, I'm having a hard time with how the locking is handled). I'll move that to the commit message to clarify that it's not only about the clobbering.